src/Importify/Main.hs

```diff
@@ -42,26 +44,36 @@ doSource src = do
     let exts = fromMaybe [] $ getExtensions moduleName extensionMaps
     let (ast, imports) = parseForImports exts src
 
-    maybe (pure "")
-          (\(start, end) -> do
-                  let codeLines        = lines src
-                  let (preamble, rest) = splitAt (start - 1) codeLines
-                  let (_, decls)       = splitAt (end - start + 1) rest
+    case importSlice imports of
+        Just (start, end) -> do
+            let codeLines        = lines src
+            let (preamble, rest) = splitAt (start - 1) codeLines
+            let (_, decls)       = splitAt (end - start + 1) rest
 
-                  unusedIds <- collectUnusedIds ast imports
-                  let newImports = removeIdentifiers unusedIds imports
+            environment <- loadEnvironment
+            let annotations = annotateModule ast environment
+            let unusedIds = collectUnusedSymbols environment imports annotations
+            let usedQuals = collectUsedQuals imports annotations
+            let newImports = cleanQuals usedQuals $ removeIdentifiers unusedIds imports
 
-                  pure $ unlines preamble
-                      <> toText (unlines $ map (toText . prettyPrint) newImports)
-                      <> unlines decls)
-        (importSlice imports)
+            pure $ unlines preamble
+                <> toText (unlines $ map (toText . prettyPrint) newImports)
+                <> unlines decls
+        Nothing -> pure src
+
+loadEnvironment :: IO Environment
+loadEnvironment = loadBase
+
+annotateModule :: Module SrcSpanInfo -> Environment -> [Scoped SrcSpanInfo]
+annotateModule ast environment =
+    let (Module l mhead mpragmas _mimports mdecls) = annotate environment ast
```

Comment:
I think it can be added into code to ease it's understanding in future. We don't have `haskell-names` experts so every piece of such information is valuable for possible debugging